### PR TITLE
Write scraper outputs to Fauna

### PR DESCRIPTION
**Pull request by tkjones117**
This PR adds logic to write each scraper's output from `scraper.js` and `scrapers_no_browser` but behind a boolean set to False. I'll turn it on after this is merged so it's a smaller PR to revert if needed.
It also includes some minor fixes, such as making a fix to the async/await logic for writing appointments.

I also added logic to read/write all the data in batches. This is actually the majority of the diff in this code. It's in preparation for when we will read the data from the tables, in batches.

### CODE REVIEW

File: lib/db-utils.js
Comment (livgust) [inline lib/db-utils.js:20]: Personally, I prefer having one key FAUNA_DB and just having it set correctly in each environment. What do you think?

File: scraper.js
Comment (livgust) [inline scraper.js:119]: So I appreciate wanting to do fewer calls but this comes at a disadvantage; CVS will have results that hang around for probably over a minute waiting for Walgreens to finish before anything is written to the DB. I really like the idea of each scraper writing its result to the DB as soon as it finishes. Are you seeing issues with that approach already? Is that why you opted to pivot to having one big call at the end?

**Review by livgust — CHANGES_REQUESTED** [review]
general q!

File: scraper.js
Comment (tkjones117) [reply scraper.js:119]: I didn't do enough research on this before deciding to batch the calls, I just assumed there would be a significant operations overhead. So I'm refactoring to have each scraper write as it's finished. I'll keep around the tests in case we do want to batch more extensively in the future.

File: lib/db-utils.js
Comment (tkjones117) [reply lib/db-utils.js:20]: That's fine with me. It just means that before I merge this, we should change the Github secret from `FAUNA_DB_DEV` and `FAUNA_DB_PROD` to just `FAUNA_DB`. And temporarily any test runs will fail.

Comment (tkjones117) [reply lib/db-utils.js:20]: wait - perhaps I misunderstood... maybe you meant keep the github secrets the same, just have one env variable. this makes sense to me.

File: scraper.js
Comment (tkjones117) [reply scraper.js]: I think this logic will actually be quite useful when we add the logic to read the data from the tables, so I think worth keeping in the code.

File: scrapers_no_browser.js
Comment (tkjones117) [inline scrapers_no_browser.js:59]: I'm going to put this behind a boolean that is set to false for now...

File: lib/db-utils.js
Comment (livgust) [inline lib/db-utils.js:12]: I might be going crazy but where is `arguments` coming from?

File: lib/db-utils.js
Comment (livgust) [inline lib/db-utils.js:32]: could we do console.error for errors instead of console.log?

File: lib/db-utils.js
Comment (livgust) [inline lib/db-utils.js:106]: do we need to worry about the massVax flag, or when we have availability but no signUpLink?

**Review by livgust — CHANGES_REQUESTED** [review]
I have a couple small nits, and I'm guessing you'll want to do a quick modification now that Rob's refactor is pushed!

Thanks so much for doing this, it looks great.

File: lib/db-utils.js
Comment (tkjones117) [reply lib/db-utils.js:12]: I grabbed this solution from here: https://javascript.plainenglish.io/lets-extend-console-log-8641bda035c3
but I'm refactoring to make this more intuitive.

File: lib/db-utils.js
Comment (tkjones117) [reply lib/db-utils.js:106]: Working on this!

File: scraper_common.js
Comment (tkjones117) [inline scraper_common.js:103]: @harcod are there any other fields I should include, that you can spot?

Comment (tkjones117) [reply scraper_common.js]: synced with Rob offline. looked in `appointmentData.service.js` and saw that I should add the `restrictions` field which I've done.

File: lib/db-utils.js
Comment (tkjones117) [reply lib/db-utils.js:106]: Added `massVax`, `siteTimestamp`, and `restrictions`. 

I think the frontend filters out entries that have availability but no signUpLink? And since I'm just trying to get the DB behaving like our current setup, I wasn't planning to remove those entries at the moment.

**Comment by tkjones117** [comment]
@livgust this is ready for re-review.

File: lib/db-utils.js
Comment (tkjones117) [reply lib/db-utils.js:106]: Oh, and I only added them to the codepaths we actually call right now (this particular function is part of the batching that we aren't calling at the moment).

**Comment by tkjones117** [comment]
Once this is merged, we just need to flip the boolean `WRITE_TO_FAUNA` to be `true`. @livgust we should decide if we want to keep writing to the dev database for the moment (to make sure it works), and then once we're confident in it, we can create the prod database and update the Github secret (`secrets.FAUNA_DB_PROD`) accordingly.

**Review by iann — APPROVED** [review]
I don't want to hold things up as this is probably a permissions thing on my end. I flipped the switch to `WRITE_TO_FAUNA` to test locally. I keep getting Unauthorized with the credentials to Fauna I have. Based on what I saw in @tkjones117 demo this morning it looks good!

**Comment by iann** [comment]
I see my run in Fauna after fixing my `FAUNA_DB` property!

File: scraper_common.js
Comment (tkjones117) [reply scraper_common.js:24]: synced offline: https://macovidvaccines.slack.com/archives/C01MZB92B8B/p1617130878120500

**Pull request merged**
